public/pages/utils/anomalyResultUtils.ts

```diff
@@ -674,13 +712,31 @@ export const parsePureAnomalies = (
   if (anomaliesHits.length > 0) {
     anomaliesHits.forEach((item: any) => {
       const rawAnomaly = get(item, '_source');
+      const featureDataMap: {[key: string]: string} = {};
+      get(rawAnomaly, 'feature_data', []).filter(element => get(element, 'feature_id') && get(element, 'feature_name')).
+      forEach((feature: any) => {
+        featureDataMap[get(feature, 'feature_id')] = get(feature, 'feature_name');
+      });
+      const relevantAttributionMap: {[key: string]: number} = {};
+      get(rawAnomaly, 'relevant_attribution', []).filter(element => get(element, 'feature_id') && get(element, 'data')).
+      forEach((attribution: any) => {
+        relevantAttributionMap[get(attribution, 'feature_id')] = toFixedNumberForAnomaly(get(attribution, 'data'));
+      });
+      const contributions: { [key: string]: FeatureContributionData } = {}
+      for (const [key] of Object.entries(featureDataMap)) {
+        contributions[key] = {
+          name: featureDataMap[key],
+          attribution: relevantAttributionMap[key]
+        }
+      }
```

Comment:
looks to be duplicate of lines 581-597 - can you combine into a helper fn? Also, prefer this version which enforces `toFixedNumberForAnomaly` on line 723

Reply:
sure, will refactor this part later